game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java

```diff
@@ -119,6 +119,19 @@ public static InGameLobbyWatcher newInGameLobbyWatcher(final IServerMessenger ga
     }
   }
 
+  @VisibleForTesting
+  static @Nullable String getLobbySystemProperty(final String key) {
+    final String backupKey = key + ".backup";
```

Comment:
@ssoloff great work finding this. Though, one RC to the bug was a lack of documentation/linking to the coupled location. It is a bit of the jenga code problem, touch one place and another falls over somewhere else. 

To this end, can we make the dependency of the property name explicit by generating the name with a common method? This would give a maintenance programmer a way to trace the dependency in case they wonder why we create backups of a system property.

If that is not a reasonable suggestion, what woudl you think of a simple comment to explain why this is here? Is there perhaps an even better approach?

Reply:
Previously, the implementation of this "feature" was spread across four different classes.  So, now that it's all co-located within a single method with associated unit tests that document the behavior, it should make future devs at least hesitate before concluding the backup properties are write-only.

I think the long-term solution to this issue is to understand the requirement that states these specific lobby system properties must be cleared after starting a game host.  For all we know, it may no longer even be applicable.  But, on the assumption it is, we can probably find a better way to handle this data by extracting it out of system properties early on in the application lifecycle and explicitly passing it downstream, where needed.

**EDIT:** I just took a quick look to see who else reads these system properties, and I can't find a single usage outside of this class that reads `LOBBY_HOST` and `LOBBY_PORT`.  There's one other read of `LOBBY_GAME_HOSTED_BY` when generating an auto-save file name, but it's only for fallback purposes (in the event another system property is not set), and that branch may never actually be hit.

So, it's quite possible, that clearing these system properties is totally unnecessary.